typing_extensions/src_py3/typing_extensions.py

```diff
@@ -1755,3 +1770,170 @@ def get_type_hints(obj, globalns=None, localns=None, include_extras=False):
         if include_extras:
             return hint
         return {k: _strip_annotations(t) for k, t in hint.items()}
+
+elif HAVE_ANNOTATED:
+
+    # Prior to Python 3.7 types did not have `copy_with`. A lot of the equality
+    # checks, argument expansion etc. are done on the _subs_tre. As a result we
+    # can't provide a get_type_hints function that strips out annotations.
+
+    class AnnotatedMeta(typing.GenericMeta):
+        """Metaclass for Annotated"""
+
+        def __new__(cls, name, bases, namespace, **kwargs):
+            if any(b is not object for b in bases):
+                raise TypeError("Cannot subclass " + str(Annotated))
+            return super().__new__(cls, name, bases, namespace, **kwargs)
+
+        @property
+        def __metadata__(self):
+            return self._subs_tree()[2]
+
+        def _tree_repr(self, tree):
+            assert len(tree) == 3
+            # First argument is this class, second is __origin__, 3rd is __metadata__
+            tp_tree = tree[1]
+            if not isinstance(tp_tree, tuple):
+                tp_repr = typing._type_repr(tp_tree)
+            else:
+                tp_repr = tp_tree[0]._tree_repr(tp_tree)
+            metadata_reprs = ", ".join(repr(arg) for arg in tree[2])
+            return repr(tree[0]) + '[%s, %s]' % (tp_repr, metadata_reprs)
+
+        def _subs_tree(self, tvars=None, args=None):
+            if self is Annotated:
+                return Annotated
+            res = super()._subs_tree(tvars=tvars, args=args)
+            # Flatten nested Annotated
+            if isinstance(res[1], tuple) and res[1][0] is Annotated:
+                sub_tp = res[1][1]
+                sub_annot = res[1][2]
+                return (Annotated, sub_tp, sub_annot + res[2])
+            return res
+
+        def _get_cons(self, msg, exc_type):
+            """ Return the class used to create instance of this type.
+
+            The msg and exc_type argument are used to control the exceptions
+            that get raised when there's no underlying constructor available (
+            e.g.: we're in an unspecialized class).
+            """
+            if self.__origin__ is None:
+                raise TypeError(msg + " a non specialized Annotated type")
+            tree = self._subs_tree()
+            while isinstance(tree, tuple) and tree[0] is Annotated:
+                tree = tree[1]
+            if isinstance(tree, tuple):
+                cons = tree[0]
+            else:
+                cons = tree
+            type_error = None
+            if isinstance(cons, typing._ForwardRef):
+                type_error = "a forward reference."
+            elif isinstance(cons, typing.TypeVar):
+                type_error = "a type variable."
+
+            if type_error:
+                error = "{} {}: the annotated type is {}".format(msg, self, type_error)
+                raise exc_type(error)
```

Comment:
TBH I think this would be cleaner if this method just returns `cons`, while the caller would raise an appropriate error and use another helper method to construct the error message.

More generally, why do we actually need this? If this method just returns a `TypeVar`, then a user will see exactly the same error as if one would try to instantiate it directly, etc.